nltk/model/counter.py

```diff
@@ -0,0 +1,99 @@
+# Natural Language Toolkit: Language Model Counters
+#
+# Copyright (C) 2001-2016 NLTK Project
+# Author: Ilia Kurenkov <ilia.kurenkov@gmail.com>
+# URL: <http://nltk.org/>
+# For license information, see LICENSE.TXT
+
+from __future__ import unicode_literals
+
+from collections import Counter, defaultdict
+from itertools import chain
+
+from nltk.util import ngrams
+from nltk.probability import FreqDist, ConditionalFreqDist
+from nltk import compat
+
+
+@compat.python_2_unicode_compatible
+class LanguageModelVocabulary(Counter):
+    """Stores language model vocabulary.
+
+    Satisfies two common language modeling requirements for a vocabulary:
+    - When checking membership and calculating its size, filters items by comparing
+      their counts to a cutoff value.
+    - Adds 1 to its size so as to account for "unknown" tokens.
+    """
+
+    def __init__(self, unknown_cutoff, *counter_args):
+        Counter.__init__(self, *counter_args)
+        self.cutoff = unknown_cutoff
+
+    def __contains__(self, item):
+        """Only consider items with counts GE to cutoff as being in the vocabulary."""
+        return self[item] >= self.cutoff
```

Comment:
I actually wonder if empty `training_data` may not be such a fringe case. Imagine this scenario:
We use different texts for creating the vocabulary (`vocab_data`) and for training (`training_data`). We'd like to be able to do something along the following lines:
```python
>>> counter = NgramCounter(highest_order=3, unk_cutoff=2)
>>> counter.build_vocabulary(vocab_data)
>>> counter.train_counts(training_data)
````